opencensus/trace/span_context.py

```diff
@@ -64,7 +58,7 @@ def __init__(
             trace_id = generate_trace_id()
 
         if trace_options is None:
-            trace_options = DEFAULT
+            trace_options = trace_options_module.TraceOptions()
```

Comment:
Re-using the default trace options here meant that tracers would set each others' contexts' sampling bit.

https://github.com/census-instrumentation/opencensus-python/blob/035a6dcbf45447468a18a2755a21fa0561384a2a/opencensus/trace/tracer.py#L84

I suspect there are other bugs like this lurking in the module.

Reply:
TODO: add a test for tracer isolation.